### PR TITLE
fix issue #1985 - FileUpload locale not work correctly

### PR DESCRIPTION
**Pull request by themahdavi**
this change for fix issue #1985 .

my solution:
1- define array of available locale filename from filepond
2- search array for find site locale
3- set the correct name to url

### CODE REVIEW

**Review by pxlrbt — CHANGES_REQUESTED** [review]
Check the styling

**Comment by themahdavi** [comment]
fix formatting on second commit (wip)

File: packages/forms/resources/views/components/file-upload.blade.php [hunk 1]
```diff
@@ -5,8 +5,12 @@
 
             $defaultLocaleData = ($placeholder = $getPlaceholder()) ? "{ labelIdle: '{$placeholder}' }" : '{}';
 
-            if (! str_contains($locale, '-')) {
-                $locale .= '-' . $locale;
+            $allAvailableLocale = ['ar-ar', 'cs-cz', 'da-dk', 'de-de', 'en-en', 'es-es', 'fa_ir', 'fr-fr', 'he-he', 'hu-hu', 'id-id', 'it-it', 'ja-ja', 'lt-lt', 'nl-nl', 'no-nb', 'pl-pl', 'pt-br', 'ru-ru', 'sk-sk', 'sv_se', 'tr-tr', 'uk-ua', 'zh-cn', 'zh-tw'];
+
+            foreach ($allAvailableLocale as $availableLocale){
+                if(str_contains($availableLocale, $locale)){
```
Comment (pxlrbt) [inline packages/forms/resources/views/components/file-upload.blade.php:11]:
```suggestion
            foreach ($allAvailableLocale as $availableLocale) {
                if (str_contains($availableLocale, $locale)) {
```

**Review by pxlrbt — CHANGES_REQUESTED** [review]

**Comment by pxlrbt** [comment]
> fix formatting on second commit (wip)
> 
> meaning about styling ?

Sorry, somehow it didn't show the requested changes 🤷🏼‍♂️ Tried it again.

Also I think we don't need to try loading the locale, if it didn't match (Lines 18–20)
```html
<script type="module">
    import localeData from 'https://cdn.skypack.dev/filepond/locale/{{$locale}}.js'
    window.FilePond && window.FilePond.setOptions({ ...localeData, ...{!! $defaultLocaleData !!} })
</script>
        ```

**Comment by themahdavi** [comment]
this pr just for load all available locale from filepond ... 
if not load and get error from get request haven't any idea
can use fallback_locale from laravel app.php config

**Comment by pxlrbt** [comment]
> this pr just for load all available locale from filepond ...
> if not load and get error from get request haven't any idea
> can use fallback_locale from laravel app.php config

I don't think we need to check the request. Just whether the locale is in `allAvailableLocales`. If it's not there we don't need to import it.

**Comment by themahdavi** [comment]
yep the default locale is EN and if the locale not contain in allAvailableLocale can ignore from send request.

commit now

**Comment by danharrin** [comment]
Tbh, this is more of a Filepond issue than Filament. Only 3 Filepond locales use `_` instead of `-`. Surely, we should encourage that project to fix the inconsistency, instead of introducing this extra code to Filament?

**Comment by danharrin** [comment]
Instead, maybe we should add a new translation key that allows you to define a Filepond locale string for that locale. Then we dont need that custom code, we can just read the locale from `__('forms::components.file_upload.filepond_locale')`

**Comment by pxlrbt** [comment]
> Instead, maybe we should add a new translation key that allows you to define a Filepond locale string for that locale. Then we dont need that custom code, we can just read the locale from `__('forms::components.file_upload.filepond_locale')`

I guess that would be the best solution 😅

**Pull request merged**